cpp/cmake/common.cmake

```diff
@@ -1,8 +1,21 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.24)
 
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
+# We use these flags over presets to avoid having to create a CMakePreset.json file in every project.
+# If you want to enable these, set this option to OFF: -DICE_DEMO_WARNINGS=OFF
```

Comment:
You mean _disable_ these warnings?

Reply:
Yes, I went back and forth a few times on what the default should be.